src/functions.php

```diff
@@ -176,8 +176,14 @@
  * await($promise);
  * ```
  *
- * @param callable $function
- * @return callable(mixed ...): PromiseInterface<mixed>
+ * @template T
+ * @template A1 (any number of function arguments, see https://github.com/phpstan/phpstan/issues/8214)
```

Comment:
"640K ought to be enough for anybody."

See https://github.com/phpstan/phpstan/issues/8214 for longer-term solution once supported in PHPStan.